TicketBooking/Controllers/BookingController.cs

```diff
@@ -0,0 +1,67 @@
+using Microsoft.AspNetCore.Mvc;
+using TicketBooking.Helpers;
+using TicketBooking.Models;
+
+namespace TicketBooking.Controllers
+{
+    [ApiController]
+    [Route("[controller]")]
+    public class BookingController : ControllerBase
+    {
+        private readonly ILogger<BookingController> _logger;
+        private readonly ISeatLockProvider seatLockProvider;
+
+        public BookingController(ILogger<BookingController> logger, ISeatLockProvider seatLockProvider)
+        {
+            _logger = logger;
+            this.seatLockProvider = seatLockProvider;
+        }
+
+        [HttpGet]
+        public IActionResult Post(BookingRequest bookingRequest)
+        {
+            try
+            {
+                var customerData = CustomerDB.Get(bookingRequest.CustomerId);
+
+                if (bookingRequest.RequestDate == null)
```

Comment:
This is an incorrect validation. DateTime cannot be null as it is a struct and not a class